More/EAV/Attribute.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace DesignPatterns\More\EAV;
+
+/**
+ * Class Attribute
+ */
+class Attribute implements ValueAccessInterface
+{
+    /**
+     * @var array|Value[]|ValueInterface[]
+     */
+    private $values = array();
+
+    /**
+     * @var string
+     */
+    private $name;
+
+    /**
+     * @return array|Value[]|ValueInterface[]
+     */
+    public function getValues()
+    {
+        return $this->values;
+    }
+
+    /**
+     * @param Value|ValueInterface $value
+     * @return $this
+     */
+    public function addValue(ValueInterface $value)
+    {
+        // @TODO I think the $value should be checked for uniqueness first to avoid duplication in array.
+        $this->values[] = $value;
+
+        return $this;
+    }
+
+    /**
+     * @param Value|ValueInterface $value
```

Comment:
There are a few places this occurs, but here the **Typehint** on the method is clearly `ValueInterface`